master/internal/webhooks/api_webhook.go

```diff
@@ -3,16 +3,40 @@ package webhooks
 import (
 	"context"
 
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"github.com/determined-ai/determined/master/internal/grpcutil"
+
 	"github.com/determined-ai/determined/proto/pkg/apiv1"
 )
 
 // WebhooksAPIServer is an embedded api server struct.
 type WebhooksAPIServer struct{}
 
+// AuthorizeRequest checks if the user has CanEditWebhooks permissions.
+func AuthorizeRequest(ctx context.Context) error {
+	curUser, _, err := grpcutil.GetUser(ctx)
+	if err != nil {
+		return status.Errorf(codes.Internal, "failed to get the user: %s", err)
+	}
+	if ok, authErr := AuthZProvider.Get().
+		CanEditWebhooks(curUser); authErr != nil {
+		return authErr
+	} else if !ok {
+		return status.Error(codes.PermissionDenied, err.Error())
```

Comment:
this ```err``` will always be nil here since it is the one from the ```grpc.GetUser```

I think it is fine to make ```AuthZProvider.Get().CanEditWebhooks``` only return a error then if that error is non nil return it as ```status.Error(codes.PermissionDenied, err.Error())```  

Just for some context why some ```authz``` interfaces returns booleans and some just returns errors, ```GetExperiment``` returns ```(canGetExperiment, serverError)``` because we want to return a 404 error when a user can't view an experiment to avoid people finding out about experiments that don't exist. The ```serverError``` is just in case some db failure occurs. In this case since we don't need to return a 404 when someone can't ```EditWebHooks``` since it is a all or nothing check either you can edit every webhook or none.

Reply:
Thanks for the info and suggestion! I have updated based on this feedback.